astropy/visualization/mpl_normalize.py

```diff
@@ -226,3 +228,61 @@ def simple_norm(data, stretch='linear', power=1.0, asinh_a=0.1, min_cut=None,
     vmin, vmax = interval.get_limits(data)
 
     return ImageNormalize(vmin=vmin, vmax=vmax, stretch=stretch, clip=clip)
+
+
+# used in imshow_norm
+_norm_sig = inspect.signature(ImageNormalize)
+
+
+def imshow_norm(data, ax=None, **kwargs):
+    """ A convenience function to call matplotlib's `matplotlib.pyplot.imshow`
+    function, using an `ImageNormalize` object as the normalization.
+
+    Parameters
+    ----------
+    data
+        The data to show. Can be whatever type `matplotlib.pyplot.imshow` and
+        `ImageNormalize` accept.
+    ax : None or `~matplotlib.axes.Axes`
+        If None, use pyplot's imshow.  Otherwise, calls ``imshow`` method of the
+        supplied axes.
+
+    All other keyword arguments are parsed first by the `ImageNormalize`
+    initializer, then to ``imshow``.
+
+    Notes
+    -----
+    The ``data`` matplotlib keyword is not supported.
+    """
+    if 'X' in kwargs:
+        raise ValueError('Cannot give both ``X`` and ``data``')
+
+    if 'norm' in kwargs:
+        raise ValueError('There is not point in using imshow_norm if you give '
+                         's ``norm`` keyword - use imshow directly if you want '
+                         'that.')
+
+    imshow_kwargs = dict(kwargs)
+
+    norm_kwargs = {'data': data}
+    for pname in _norm_sig.parameters:
+        if pname in kwargs:
+            norm_kwargs[pname] = imshow_kwargs.pop(pname)
+
+    # now re-assign any "_mpl" kweywords to their equivalents
+    for pname in tuple(imshow_kwargs):
+        if pname.endswith('_mpl'):
```

Comment:
What are these `*_mpl` keywords?  Why are they needed?

Reply:
See my response in https://github.com/astropy/astropy/pull/7785#discussion_r227209274